tests/components/devolo_home_control/mocks.py

```diff
@@ -425,3 +427,30 @@ def __init__(self, **kwargs: Any) -> None:
         self.devices = {"Test": SwitchMock()}
         self.publisher = Publisher(self.devices.keys())
         self.publisher.unregister = MagicMock()
+
+
+class MydevoloMock(Mydevolo):
+    """Mydevolo mock."""
+
+    def __init__(self) -> None:
+        """Initialize the mock."""
+        super().__init__()
+        self.valid_credentials = True
+        self.in_maintenance = False
+
+    def credentials_valid(self) -> bool:
+        """Return True to simulate valid credentials."""
+        return self.valid_credentials
+
+    def maintenance(self) -> bool:
+        """Return False to simulate no maintenance."""
+        return self.in_maintenance
+
+    def get_gateway_ids(self) -> list[str]:
+        """Return a list of gateway IDs."""
+        return ["1400000000000001", "1400000000000002"]
+
+    @lru_cache(maxsize=1)
+    def uuid(self) -> str:
+        """Return a mock UUID."""
+        return "123456"
```

Comment:
I am wondering what the benefit is of extending the real class over just using an AsyncMock to patch out the fields

Reply:
I didn't see a way to change the mock at runtime of the tests so I chose the same approach I used in the devolo Home Network integration. I need to change the mock to be able to test recovery of errors.